pkg/endpoint/metrics.go

```diff
@@ -0,0 +1,61 @@
+package endpoint
+
+import (
+	"math"
+	"time"
+
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	"github.com/cilium/cilium/pkg/metrics"
+	"github.com/cilium/cilium/pkg/spanstat"
+)
+
+type regenerationStatistics struct {
+	success                bool
+	totalTime              spanstat.SpanStat
+	waitingForLock         spanstat.SpanStat
+	waitingForCTClean      spanstat.SpanStat
+	policyCalculation      spanstat.SpanStat
+	proxyConfiguration     spanstat.SpanStat
+	proxyPolicyCalculation spanstat.SpanStat
+	proxyWaitForAck        spanstat.SpanStat
+	bpfCompilation         spanstat.SpanStat
+	mapSync                spanstat.SpanStat
+	prepareBuild           spanstat.SpanStat
+}
+
+// SendMetrics get the curret statistics and push the info to the given prometheus metrics.
```

Comment:
"SendMetrics sends the regeneration statistics for this endpoint to Prometheus."